conans/model/conan_package.py

```diff
@@ -0,0 +1,94 @@
+import os
+
+class ConanPackage(object):
+    """
+    models conan_package.json, serializable object
+    """
+
+    DEFAULT_FILENAME = os.path.join(".autopackage", "conan_info.json")
+
+    def __init__(self):
+        self.configurations = dict()
+
+    @classmethod
+    def load(cls, filename=DEFAULT_FILENAME):
+        """
+        loads package model into memory from the conan_package.json file
+        :param filename: path to the conan_package.json
+        :return: a new instance of ConanPackage
+        """
+        import json
+        from conans.util.files import load
+        def from_json(o):
+            if "configurations" in o:
+                return {n: ConanPackage.Configuration(c) for n, c in o["configurations"].items()}
+            elif "components" in o:
+                return {n: ConanPackage.Component(c) for n, c in o["components"].items()}
+            else:
+                return o
+        conan_package = ConanPackage()
+        conan_package.configurations = json.loads(load(filename), object_hook=from_json)
+        return conan_package
+
+    def save(self, filename=DEFAULT_FILENAME):
+        """
+        saves package model from memory into the conan_package.json file
+        :param filename: path to the conan_package.json
+        :return: None
+        """
+        import json
+        from conans.util.files import save
+        save(filename, json.dumps(self, default=lambda o: o.__dict__))
+
+    def package_info(self, conanfile):
+        """
+        performs an automatically generated package_info method on conanfile, populating
+        conanfile.package_info with the information available inside conan_package.json
+        :return: None
+        """
+        build_type = conanfile.settings.get_safe("build_type")
+        configuration = self.configurations[build_type]
+        for cname, component in configuration.components.items():
+            conanfile.cpp_info.components[cname].libs = component.libs
+            conanfile.cpp_info.components[cname].requires = component.requires
+            for generator, gname in component.names.items():
+                conanfile.cpp_info.components[cname].names[generator] = gname
+
+    def add_configuration(self, name):
+        """
+        appens a new ConanPackage.Configuration into the internal dictionary
+        :param name: name of the given configuration (e.g. Debug)
+        :return: a new ConanPackage.Configuration instance (empty)
+        """
+        self.configurations[name] = ConanPackage.Configuration()
+        return self.configurations[name]
+
+    class Configuration(object):
```

Comment:
Package descriptions should be configuration agnostic, please do not include specific Debug/Release things.